timer.r/scripts/split.r

```diff
@@ -0,0 +1,176 @@
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(tictoc)
+library(knitr)
+
+
+tic()
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+toc()
+
+
+# Assignment 2:  
+
+tic()
+MTweedieTests <-  
+  function(N,M,sig){ 
+    sum(replicate(M,simTweedieTest(N)) < sig)/M 
+  } 
+toc()
+
+# Assignment 3:  
+#Load package
+library(doParallel)
+
+# Parallelized MTweedieTests
+MTweedieTests_parallel <- function(N, M, sig){
+  results <- foreach(m = 1:M, .combine = c, .packages = "tweedie") %dopar% {
+    simTweedieTest(N) < sig
+  }
+  
+  return(sum(results) / M)
+}
+
+
+# Register the number of cores to be used for parallel computation
+# This example uses 8 cores. 
+maxcores <- 8
+Cores <- min(parallel::detectCores(), maxcores)
+cl <- makeCluster(Cores)
+registerDoParallel(cl)
+
+
+
+clusterExport(cl, list("MTweedieTests_parallel", "simTweedieTest"))
+
+df <-  
+  expand.grid( 
+    N = c(10,100,1000,5000, 10000), 
+    M = 1000, 
+    share_reject = NA) 
+
+tic(paste0("Parallel loop for MTweedieTests, ", Cores, " cores"))
+for(i in 1:nrow(df)){ 
+  df$share_reject[i] <-  
+    MTweedieTests_parallel( 
+      N=df$N[i], 
+      M=df$M[i], 
+      sig=.05) 
+} 
+toc()
+
+# Close the cluster once done
+stopCluster(cl)
```

Comment:
Results are correct. However it would be cleaner if you included all the parallell code lines such as the lines from44-51 and 70 inside the function.